Generator/AttributeOptionGenerator.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace Pim\Bundle\DataGeneratorBundle\Generator;
+
+use Faker;
+use Pim\Bundle\CatalogBundle\Repository\AttributeRepositoryInterface;
+use Pim\Bundle\CatalogBundle\Repository\LocaleRepositoryInterface;
+use Symfony\Component\Console\Helper\ProgressHelper;
+use Symfony\Component\Yaml;
+
+/**
+ * Generate native YML file for attribute option useable as fixtures
+ *
+ * @author    Claire Fortin <claire.fortin@akeneo.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class AttributeOptionGenerator implements GeneratorInterface
+{
+    const ATTRIBUTE_OPTION_CODE_PREFIX = 'attr_opt_';
+
+    const DEFAULT_FILENAME = 'attribute_options.yml';
+
+    /** @var string */
+    protected $outputFile;
+
+    /** @var string */
+    protected $delimiter;
+
+    /** @var array */
+    protected $locales;
+
+    /** @var array */
+    protected $attributeOptions = [];
+
+    /** @var LocaleRepositoryInterface */
+    protected $localeRepository;
+
+    /** @ var AttributeRepositoryInterface */
+    protected $attributeRepository;
+
+    /** @var array */
+    protected $attributes;
+
+    /** @var Faker\Generator */
+    protected $faker;
+
+    /**
+     * @param LocaleRepositoryInterface    $localeRepository
+     */
+    public function __construct(
+        LocaleRepositoryInterface $localeRepository
+    ) {
+        $this->localeRepository   = $localeRepository;
+        $this->faker = Faker\Factory::create();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generate(array $config, $outputDir, ProgressHelper $progress, array $options = null)
+    {
+        $count = (int) $config['count'];
+
+        $this->outputFile = (!empty($config['filename'])) ?
```

Comment:
If the ternary operator is too large to be on a single line, it definitely means that it should be replace by a if ;)